src/scheduler.cc

```diff
@@ -9,23 +9,71 @@
 #include <dirent.h>
 #include <dlfcn.h>
 #include <link.h>
-#include "rccl/rccl.h"
+#include <string>
+#include <curl/curl.h>
+#include <nlohmann/json.hpp>
+
+#ifdef RCCL
+  #include "rccl/rccl.h"
+#else 
+  #include "nccl.h"
+#endif
 #include "parser.h"
 
 #define __hidden __attribute__ ((visibility("hidden")))
 
-#define MSCCL_SCHEDULER_NAME "github.com/microsoft/msccl-scheduler"
+#define MSCCL_SCHEDULER_NAME "github.com/Azure/msccl-scheduler"
 
 static const char* mscclAlgoDirEnv = "MSCCL_ALGO_DIR";
 static const char* mscclAlgoDefaultDir = "msccl-algorithms";
 extern "C" bool mscclUnitTestMode() __attribute__((__weak__));
 static const char* mscclUnitTestAlgoDefaultDir = "msccl-unit-test-algorithms";
-static const char* mscclAlgoShareDirPath = "share/rccl/msccl-algorithms";
-static const char* mscclUnitTestAlgoShareDirPath = "share/rccl/msccl-unit-test-algorithms";
+static const char* mscclAlgoShareDirPath = "share/msccl-scheduler/msccl-algorithms";
+static const char* mscclUnitTestAlgoShareDirPath = "..share/msccl-unit-test-algorithms";
+static const char* mscclAzureVMDetectionAgent = "http://169.254.169.254/metadata/instance?api-version=2019-06-04";
 
 static std::vector<mscclAlgoMeta> mscclAlgoMetas;
 static std::vector<std::map<int, mscclAlgoHandle_t>> rankToAlgoHandles;
 
+static size_t writeCallback(void* contents, size_t size, size_t nmemb, std::string* userp) {
+    userp->append((char*)contents, size * nmemb);
+    return size * nmemb;
+}
+
+static std::string updateAlgoDirByVMSize(std::string algoDir){
+    CURL* curl;
+    CURLcode res;
+    std::string readBuffer;
+    std::string vmSize;
+    std::string updatedAlgoDir = algoDir;
+
+    curl_global_init(CURL_GLOBAL_DEFAULT);
+    curl = curl_easy_init();
+    if(curl) {
+        curl_easy_setopt(curl, CURLOPT_URL, mscclAzureVMDetectionAgent);
+        struct curl_slist *headers = NULL;
+        headers = curl_slist_append(headers, "Metadata:true");
+        curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeCallback);
+        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &readBuffer);
+        curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT, 1L);
+        res = curl_easy_perform(curl);
+        if(res != CURLE_OK) {
+            fprintf(stderr, "MSCCL Get Azure VM Size failed: %s", curl_easy_strerror(res));
```

Comment:
Logs from MSCCL scheduler should use its own tag, line 90 is a good example (fprintf(stderr, "%s: dladdr1 failed", MSCCL_SCHEDULER_NAME);)

Reply:
changed accordingly.